sources/api/host-containers/src/main.rs

```diff
@@ -434,22 +434,48 @@ where
     Ok(())
 }
 
+fn is_restart_required(changed_settings: &str, container_name: &str) -> bool {
+    if changed_settings.is_empty() {
+        // it means that Bottlerocket is booting - all containers need to be started
+        return true;
+    }
+
+    let settings: Vec<&str> = changed_settings.split_whitespace().collect();
+    let setting_prefix = format!("host-containers.{}", container_name);
+
+    for setting in settings.iter() {
+        if setting.contains(&setting_prefix) {
+            info!("Found a changed setting for {} container", container_name);
+            return true;
+        }
+        if !setting.contains("host-containers") {
```

Comment:
```suggestion
        if !setting.starts_with(&setting_prefix) {
```
(if we split up the settings, per previous suggestion)